cpp/src/arrow/gpu/cuda_context.cc

```diff
@@ -301,8 +328,9 @@ Status CudaContext::OpenIpcBuffer(const CudaIpcMemHandle& ipc_handle,
 
 Status CudaContext::CloseIpcBuffer(CudaBuffer* buf) {
   ContextSaver set_temporary(reinterpret_cast<CUcontext>(handle()));
-  CU_RETURN_NOT_OK(
-      cuIpcCloseMemHandle(reinterpret_cast<CUdeviceptr>(buf->mutable_data())));
+  auto data = reinterpret_cast<CUdeviceptr>(buf->mutable_data());
+  if (data != 0)  // non 0-size buffer, see ExportIpcBuffer
```

Comment:
this should probably also be `data != reinterpret_cast<CUdeviceptr>(0)`

Reply:
That would not work either. See above.